src/YesSql.Core/Store.cs

```diff
@@ -82,8 +91,26 @@ internal Store(IConfiguration configuration) : this()
             Dialect = Configuration.SqlDialect;
         }
 
+        /// <summary>
+        /// Initializes a <see cref="Store"/> instance using a specific <see cref="Configuration"/> instance.
+        /// </summary>
+        /// <param name="configuration">The <see cref="Configuration"/> instance to use.</param>
+        internal Store(IConfiguration configuration, string[] collections) : this(configuration)
+        {
+            _collections.AddRange(collections);
+        }
+
         public async Task InitializeAsync()
         {
+            await _semaphore.WaitAsync();
+
+            if (_isInitialized)
+            {
+                _semaphore.Release();
+
+                return;
+            }
```

Comment:
I meant something like this

            if (_isInitialized)
            {
                return;
            }

            await _semaphore.WaitAsync();
            try
            {
                if (_isInitialized)
                {
                    return;
                }

                ...
                ...
                _isInitialized = true;
            }
            finally
            {
                _semaphore.Release();
            }

Reply:
Done.